src/Motor.Extensions.Hosting.RabbitMQ/RabbitMQMessageConsumer.cs

```diff
@@ -122,34 +131,44 @@ private void ConsumerCallback(BasicDeliverEventArgs args)
             var task = ConsumeCallbackAsync?.Invoke(cloudEvent, _stoppingToken)?
                 .ConfigureAwait(false)
                 .GetAwaiter();
-            task?.OnCompleted(() =>
+            task?.OnCompleted(async () =>
```

Comment:
I'm not quite sure this is the best solution. `OnCompleted` takes an `Action`, which is not `async` and it doesn't look like there's an async overload.

Reply:
I reconsidered my decision and came up with a different solution. Would you please have a look @FlorianHockmann?

Reply:
Your new solution looks way cleaner to me and I don't see why it shouldn't work or whether there was any special reason for using this `OnCompleted` solution other than missing support for async/await in the RabbitMQ library.